core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -478,7 +478,8 @@ private[spark] class BlockManager(
     }
 
     hostLocalDirManager =
-      if (conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED)) {
+      if (conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED) &&
+        !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)) {
```

Comment:
cc @vanzin

Reply:
Shall we update the following configuration description by mentioning `SHUFFLE_USE_OLD_FETCH_PROTOCOL`?
```
private[spark] val SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED =
     ConfigBuilder("spark.shuffle.readHostLocalDisk.enabled")
       .doc("If enabled, shuffle blocks requested from those block managers which are running on " +
         "the same host are read from the disk directly instead of being fetched as remote blocks " +
         "over the network.")
       .booleanConf
       .createWithDefault(true)
```